packages/playwright-core/src/server/registry/browserFetcher.ts

```diff
@@ -49,14 +48,8 @@ export async function downloadBrowserWithProgressBar(title: string, browserDirec
       debugLogger.log('install', `attempt #${attempt} - ERROR: ${errorMessage}`);
       if (attempt >= retryCount)
         throw error;
-    }
-    debugLogger.log('install', `extracting archive`);
-    debugLogger.log('install', `-- zip: ${zipPath}`);
-    debugLogger.log('install', `-- location: ${browserDirectory}`);
-    await extract(zipPath, { dir: browserDirectory });
-    if (executablePath) {
-      debugLogger.log('install', `fixing permissions at ${executablePath}`);
-      await fs.promises.chmod(executablePath, 0o755);
+      if (await existsAsync(zipPath))
```

Comment:
upper code got moved into out-of-process. The additional check in line 51 got added for safety precautions, since for a retry the file could potentially exist.